sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -266,8 +266,31 @@ case class Invoke(
   override def nullable: Boolean = targetObject.nullable || needNullCheck || returnNullable
   override def children: Seq[Expression] = targetObject +: arguments
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  override def eval(input: InternalRow): Any = {
```

Comment:
I basically have the same comments here as for `StaticInvoke`: Why do we need to resolve the method at runtime?  Why are you using `CodeGenerator.defaultValue(dataType)` to determine if something is a primitive?

Reply:
O, btw: We should reuse the method resolution logic in `StaticInvoke`

Reply:
Since `obj` should be evaluated at each call of `eval`, it is hard to reuse method resolution logic in `StaticInvoke`. On the other hand, we can reuse logic for casting a return value.

Reply:
I refactored the code as possible as the common code is used.